scripts/eurostat/health_determinants/alcohol_consumption/process.py

```diff
@@ -0,0 +1,820 @@
+            ,value_name='observation')
+    return df
+
+#hazardousalcoholconsumption_by_sex_urbanisation
+def hazardousalcoholconsumption_by_sex_urbanisation(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Cleans the file hazardousalcoholconsumption_by_sex_urbanisation for concatenation in Final CSV
+    Args:
+        df (pd.DataFrame): The raw df as the input
+    Returns:
+        df (pd.DataFrame): provides the cleaned df as output
+    """
+    # providing column names
+    columns = ['deg_urb,sex,age,unit,time','EU27_2020','EU28','BE','BG',
+    'CZ','DK','DE','EE','IE','EL','ES','HR','IT','CY','LV','LT','LU','HU',
+    'MT','AT','PL','PT','RO','SI','SK','FI','SE','IS','NO','UK','TR']
+    df.columns=columns
+    # spliting the first column to multiple columns
+    split_columns = "deg_urb,sex,age,unit,time"
+    df = _split_column(df,split_columns)
+    # Filtering out the wanted rows and columns
+    df = df[df['age'] == 'TOTAL']
+    df.drop(columns=['EU27_2020','EU28'],inplace=True)
+    df = _replace_deg_urb(df)
+    df = _replace_sex(df)
+    # giving proper statvar name
+    df['SV'] = 'Percent_HazardousAlcoholConsumption_In_Count_Person_'\
+        +df['deg_urb']+'_'+df['sex']
+    # dropping unwanted columns
+    df.drop(columns=['deg_urb','sex','unit','age'],inplace=True)
+    # ask
+    df = df.melt(id_vars=['SV','time'], var_name='geo'\
+        ,value_name='observation')
+    return df
+
+#alcoholconsumption_by_sex_country_of_birth
+def alcoholconsumption_by_sex_country_of_birth(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Cleans the file alcoholconsumption_by_sex_country_of_birth for concatenation in Final CSV
+    Args:
+        df (pd.DataFrame): The raw df as the input
+    Returns:
+        df (pd.DataFrame): provides the cleaned df as output
+    """
+    # providing column names
+    columns = ['unit,frequenc,sex,age,c_birth,time','EU27_2020','EU28','BE','BG',
+    'CZ','DK','DE','EE','IE','EL','ES','HR','IT','CY','LV','LT','LU','HU',
+    'MT','AT','PL','PT','RO','SI','SK','FI','SE','IS','NO','UK','TR']
+    df.columns=columns
+    # spliting the first column to multiple columns
+    split_columns = "unit,frequenc,sex,age,c_birth,time"
+    df = _split_column(df,split_columns)
+    # Filtering out the wanted rows and columns
+    df = df[df['age'] == 'TOTAL']
+    df.drop(columns=['EU27_2020','EU28'],inplace=True)
+    df = _replace_frequenc(df)
+    df = _replace_sex(df)
+    df = _replace_c_birth(df)
+    # giving proper statvar name
+    df['SV'] = 'Percent_AlcoholConsumption_'+df['frequenc']+\
+        '_In_Count_Person_'+df['sex']+'_'+df['c_birth']
+    # dropping unwanted columns
+    df.drop(columns=['frequenc','c_birth','sex','unit','age'],inplace=True)
+    # ask
+    df = df.melt(id_vars=['SV','time'], var_name='geo'\
+        ,value_name='observation')
+    return df
+
+#alcoholconsumption_by_sex_citizen
+def alcoholconsumption_by_sex_citizen(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Most of the function has similar statement, can it be generalised to single function with parameterization. (E.g., alcoholconsumption_by_sex_country_of_birth and alcoholconsumption_by_sex_citizen is very much simmilar.)